onboarding_auth/onboardingAuthPresentation/src/main/java/com/alexafit/onboardingAuthPresentation/register/RegisterScreen.kt

```diff
@@ -1,16 +1,282 @@
+                            color = MaterialTheme.colors.onSurface,
+                            modifier = Modifier
+                        )
+                    }
+                ) {
+                    if (viewModel.registerState.validEmailAddress) {
+                        Icon(
+                            painter = painterResource(R.drawable.baseline_check_24),
+                            contentDescription = stringResource(
+                                id = R.string.content_desc_image_vector
+                            ),
+                            tint = SuccessGreen,
+                            modifier = Modifier
+                                .height(IntrinsicSize.Min)
+                                .width(IntrinsicSize.Min)
+                        )
+                    }
+                }
+                Spacer(modifier = Modifier.height(spacing.spaceMedium))
+                TextFieldWithIcon(
+                    text = viewModel.registerState.password,
+                    keyboardImeAction = if (viewModel.registerState.validEmailAddress && viewModel.registerState.validName) ImeAction.Done else ImeAction.Next,
+                    keyboardType = KeyboardType.Password,
+                    modifier = Modifier.fillMaxWidth(),
+                    visualTransformation = if (viewModel.registerState.isPasswordVisible) {
+                        VisualTransformation.None
+                    } else {
+                        PasswordVisualTransformation(
+                            '●'
+                        )
+                    },
+                    onKeyboardActionPressed = {
+                        if (viewModel.registerState.validEmailAddress && viewModel.registerState.validName) {
+                            keyboardController?.hide()
+                            viewModel.onEvent(RegisterUserEvent.NavigateToAgenda)
+                        }
+                    },
+                    onValueChange = { viewModel.onEvent(RegisterUserEvent.OnPasswordEnter(it)) },
+                    onFocusChanged = { viewModel.onEvent(RegisterUserEvent.OnPasswordFocusChange(it.isFocused)) },
+                    placeholder = {
+                        Text(
+                            text = stringResource(id = R.string.text_field_hint_password),
+                            style = MaterialTheme.typography.body1,
+                            fontWeight = FontWeight.Light,
+                            color = MaterialTheme.colors.onSurface,
+                            modifier = Modifier
+                        )
+                    }
+                ) {
+                    IconButton(
+                        onClick = { viewModel.onEvent(RegisterUserEvent.OnPasswordIconClicked(viewModel.registerState.isPasswordVisible)) },
+                        modifier = Modifier
+                            .height(IntrinsicSize.Min)
+                            .width(IntrinsicSize.Min)
```

Comment:
I'd avoid using `IntrinsicSize.Min` for icon buttons, since they intentionally contain outer spacing to make the clickable area bigger. Otherwise, it might be really frustrating for users with thick fingers to locate the few pixels to click on 😄